lib/types.ts

```diff
@@ -194,3 +195,19 @@ export function callVariadicCallback<T>(
 }
 
 export type Callback<T> = CallbackNoError<T> | CallbackWithError<T>;
+
+export type InternalBreadcrumb = {
```

Comment:
I find the name of the type slightly misleading. `InternalBreadcrumb` to me sounds more that there's somewhat of a provider-internal and a provider-external breadcrumb. I would probably name this more along the lines of `Breadcrumb` (the Partial) and `BreadcrumbInternals` oder `BreadcrumbDetails` to focus more on the fact that both are around the same type in the grand scheme of things.